<!DOCTYPE html>
<html lang="en">
<head>

    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Document</title>
   
</head>

<body>

<script>

// const myPromise=new Promise((resolve,reject) => {
//     setTimeout(()=>{
//     const x=Math.random();
//     if(x < 0.5){
//         resolve("Data has been successfully retrieved.");
//     }else {
//         reject("An error occurred while fetching data.");
//     }
    
//     },2000);
// })
// myPromise
//         .then((data)=>{
//           console.log(data);
//         })
//         .catch((error)=>{
//         console.log(error);
//         }); 
        

  


//     const promise1 = Promise.resolve(123);
//     const promise2 = 456;
//     const promise3 = new Promise((resolve, reject) => {    
//     setTimeout(resolve, 1000, 'foo');
// });
    
//     Promise.all([promise1, promise2, promise3]).then((values) => {
//       console.log(values);
//     });

// Async/Await = Async = makes a function return a promise
//                           Await = makes an async function wait for a promise

// Allows you write asynchronous code in a synchronous manner
// Async doesn't have resolve or reject set up as parameters
// Everything after Await is placed in an event queue

function walkDog(){
    return new Promise((resolve, reject) => {
        setTimeout(() => {

            const dogWalked = true;

            if(dogWalked){
                resolve("You walk the dog ");
            }
            else{
                reject("You DIDN'T walk the dog");
            }
        }, 1500);
    });
}

function cleanKitchen(){
    return new Promise((resolve, reject) => {
        setTimeout(() => {
            
            const kitchenCleaned = true;

            if(kitchenCleaned){
                resolve("You clean the kitchen ");
            }
            else{
                reject("You DIDN'T clean the kitchen");
            }
        }, 2500);
    });
}

function takeOutTrash(){
    return new Promise((resolve, reject) => {
        setTimeout(() => {

            const trashTakenOut = false;

            if(trashTakenOut){
                resolve("You take out the trash ");
            }
            else{
                reject("You DIDN'T take out the trash");
            }

        }, 500);
    });
}
/*
walkDog().then((value)=>{console.log(value);return cleanKitchen()})
         .then((value)=>{console.log(value);return takeOutTrash()})
         .then((value)=>{console.log(value);console.log("You have finished all the task")})
         .catch((error)=>consloe.error(error));                                                           */

async function doChores(){
  try{
  const walkDogResult=await walkDog();
  console.log(walkDogResult);
  const cleanKitchenResult=await cleanKitchen();
  console.log(cleanKitchenResult);
  const takeOutTrashResult=await takeOutTrash();
  console.log(takeOutTrashResult);
  console.log("You have finished all the task"); 

  }
  catch(error){
    console.error(error);
  }
  


}         
doChores();    






</script>



</body>
</html>